### DISCUSSION TITLE
An Advanced Chat App with Large Language Models

### DISCUSSION THREAD

**Discussion by alikalik9** [zauberzeug/nicegui#2116] [Show and tell]
Hello everyone,

I'd like to share one of my projects related to NiceGUI with you. I took the AI chat example and significantly expanded it. The result is a chat app that allows you to converse with various large language models. 
<img width="1440" alt="Bildschirm­foto 2023-12-03 um 16 21 22" src="https://github.com/zauberzeug/nicegui/assets/126254607/e6a36e50-ae0b-4f81-bc27-3030f09bad83">

Some of the features include:

- Simple chat history
- Embeddings
- Selection of different models

You can find more details in my [Repository](https://github.com/alikalik9/openllm)

This is my most complex programming project so far, so there's definitely room to improve the efficiency/structure of the code. Nevertheless, I learned a lot while programming. Thanks to the NiceGUI team for providing this platform. Now I can give my ideas a great web UI.

**Comment by rodja** [comment]
Very nice! Thanks for sharing. I've added a link to https://github.com/zauberzeug/nicegui/wiki#community-projects.

**Comment by lovedeepsinghrj** [comment]
Nice project! I have been working on something similar. Working on some issues and improving a few things after reviewing yours. Thanks for sharing.

**Comment by liujiayi6v** [comment]
i'm very very very like nicegui~~~~
